### PR TITLE
feat!: generate docs from code

### LINKED ISSUE #201: Generate docs with go generator
It is now possible to generate provider docs using the https://www.terraform.io/docs/registry/providers/docs.html#generating-documentation generator and fields embedded in the provider.

Can we generate all of our documentation identically in this way?

The generator is noted to have some weakness handling deep structures.

Some example repositories using this method:
* https://registry.terraform.io/providers/hashicorp/boundary/latest/docs
* https://registry.terraform.io/providers/paultyng/unifi/latest/docs
* https://registry.terraform.io/providers/paultyng/sql/latest/docs

### PR DESCRIPTION
**Pull request by ctreatma**
Our existing provider docs are prone to errors because we transcribe attribute descriptions, deprecations, etc. back and forth between the code and docs by hand.  By generating docs from code, we reduce the overhead of providing accurate, up-to-date docs for our customers.

Note that this is a breaking change because, prior to this PR, docs for an Equinix resource or data source were kept in a file named `equinix_<resource>.md`.  The [recommendation from Hashicorp](https://developer.hashicorp.com/terraform/registry/providers/docs#typical-structure) is to keep docs in a file named `<resource>.md` (without the `equinix_` prefix), and the `terraform-plugin-docs` tool [strictly follows this recommendation](https://github.com/hashicorp/terraform-plugin-docs/issues/347#issuecomment-2020350570).  This means that adopting docs generation would move our existing docs from, e.g. `https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_vrf` to `https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/metal_vrf`.

This PR adopts docs generation as follows:
- Run `tfplugindocs migrate` to create templates from our existing docs
- Run `tfplugindocs generate` to regenerate the docs from the newly-created templates

**NOTE** the scope of this PR is to move our docs to align with the HashiCorp's recommended structure & naming conventions and to provide [an example of updating existing templates to automatically generate resource / datasource attribute details](https://github.com/equinix/terraform-provider-equinix/pull/645/commits/37c4d067eada9646cec4e5ee04795f9a3e74a45b).  Updating specific templates is the responsibility of each engineering team.

Closes #201

### CODE REVIEW

**Comment by displague** [comment]
May be helpful to incorporate https://github.com/equinix/terraform-provider-equinix/pull/647 as part of this

**Comment by displague** [comment]
The generated markdown does not include the `subcategory` frontmatter (Metal, Fabric, ...). 

The generated example `.tf` files look promising. It should make it easier to validate that the examples are accurate. 
How are they derived? Are they embedded in the rendered documentation?

Validated examples could help with https://github.com/crossplane-contrib/provider-jet-equinix/issues/29#issuecomment-1742156459

**Comment by ctreatma** [comment]
Might as well wait for #683 before pursuing docs generation

**Comment by displague** [comment]
Could include or rebase off #685

**Comment by ctreatma** [comment]
This PR is now based on & blocked by #683.

**Comment by ctreatma** [comment]
@equinix/governor-digin-fabric @equinix/governor-ne-network-edge-engineering please take a look.  This PR does not change the contents of your documentation, although it does change the URL for some pages to match the convention recommended by Hashicorp and used by the vast majority of terraform providers (for example `/equinix_metal_device` => `/metal_device` or `/equinix_network_ssh_key` => `/network_ssh_key`; some Fabric resources already had the correct URL).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Available via [https://github.com/equinix/terraform-provider-equinix/blob/main/.
 
 Please submit change requests and / or features via [Issues](https://github.com/equinix/terraform-provider-equinix/issues). There's no guarantee it'll be changed, but you never know until you try. We'll try to add comments as soon as possible, though.
 
+All resource and data source changes must be reflected in the documentation, including new attributes and changes to existing attribtues. The documentation is generated via `make docs`.  For more detailed instructions for updating docs, see ["Documenting the provider" in the development guide](./DEVELOPMENT.md#documenting-the-provider).
```
Comment (ocobles) [inline CONTRIBUTING.md:22]: "to existing attribtues" --> "to existing attributes"

Comment (ctreatma) [reply CONTRIBUTING.md]: fixed

**Review by ocobles — APPROVED** [review]
looks pretty good! I like the docs-check approach too simple and useful

**Review by thogarty — APPROVED** [review]
All good from a Fabric perspective. Looking forward to a point where we can start reducing the number of templates to allow the process to handle generation fresh once we get a handle on it.

Thanks, Charles!

**Pull request merged**